arrow/parser.py

```diff
@@ -424,7 +424,11 @@ def _build_datetime(parts):
                 # day not given, default to 1
                 day = 1
 
-            dt = iso_to_gregorian(year, week, day)
+            date_string = f"{year}-{week}-{day}"
+
+            # %Y: year, %W: week number of the year(Monday as the first day of the week), %w: weekdate
+            dt = datetime.strptime(date_string, "%Y-%W-%w")
```

Comment:
I think we need to use the tokens for ISO 8601 weekdates here as they don't seem to be interchangeable (https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes), i.e. `"%G-%V-%u"`

Reply:
For example:

```shell
>>> from datetime import datetime
>>> dt1=datetime.strptime("2013-10-4", "%G-%V-%u")
>>> dt1
datetime.datetime(2013, 3, 7, 0, 0)
>>> dt2=datetime.strptime("2013-10-4", "%Y-%W-%w")
>>> dt2
datetime.datetime(2013, 3, 14, 0, 0)
```

Reply:
I see. I have updated the branch so that it uses the ISO 8601 tokens. Thanks!